Bump v1.8.0

calavera opened the pull request:
Scheduled to be released on Tuesday, August 4th.

RC1 will be pushed to test.docker.com momentarily along with rpms.

Please help with testing as this will be the official RC.

Commit:
pkg/archive: new utilities for copying resources

Adds TarResource and CopyTo functions to be used for creating
archives for use with the new `docker cp` behavior.

Adds multiple test cases for the CopyFrom and CopyTo
functions in the pkg/archive package.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

codemac committed:
docker ps: add fields for ordering and column selection

* api/client/ps.go: Refactor CmdPs to use a fields list of
  characters to determine which columns to print on `docker ps`
  invocation.

This adds an ability for the docker command to print the columns of
output in arbitrary order.

Signed-off-by: Jeff Mickey <j@codemac.net>

Docker-DCO-1.1-Signed-off-by: Jeff Mickey <j@codemac.net>

Commit:
Add and modify tests for legacy and new daemon invokations

Signed-off-by: Shishir Mahajan <shishir.mahajan@redhat.com>
Signed-off-by: Tibor Vass <tibor@docker.com>

calavera committed:
Docker ps custom formatting.

Docker-DCO-1.1-Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
cli: new daemon command and new cli package

This patch creates a new cli package that allows to combine both client
and daemon commands (there is only one daemon command: docker daemon).

The `-d` and `--daemon` top-level flags are deprecated and a special
message is added to prompt the user to use `docker daemon`.

Providing top-level daemon-specific flags for client commands result
in an error message prompting the user to use `docker daemon`.

This patch does not break any old but correct usages.

This also makes `-d` and `--daemon` flags, as well as the `daemon`
command illegal in client-only binaries.

Signed-off-by: Tibor Vass <tibor@docker.com>

ewindisch committed:
Move AppArmor policy to contrib & deb packaging

The automatic installation of AppArmor policies prevents the
management of custom, site-specific apparmor policies for the
default container profile. Furthermore, this change will allow
a future policy for the engine itself to be written without demanding
the engine be able to arbitrarily create and manage AppArmor policies.

- Add deb package suggests for apparmor.
- Ubuntu postinst use aa-status & fix policy path
- Add the policies to the debian packages.
- Add apparmor tests for writing proc files
Additional restrictions against modifying files in proc
are enforced by AppArmor. Ensure that AppArmor is preventing
access to these files, not simply Docker's configuration of proc.
- Remove /proc/k?mem from AA policy
The path to mem and kmem are in /dev, not /proc
and cannot be restricted successfully through AppArmor.
The device cgroup will need to be sufficient here.
- Load contrib/apparmor during integration tests
Note that this is somewhat dirty because we
cannot restore the host to its original configuration.
However, it should be noted that prior to this patch
series, the Docker daemon itself was loading apparmor
policy from within the tests, so this is no dirtier or
uglier than the status-quo.

Signed-off-by: Eric Windisch <eric@windisch.us>

ewindisch committed:
Add AppArmor policy for the engine

Wraps the engine itself with an AppArmor policy.

This restricts what may be done by applications
we call out to, such as 'xz'.

Significantly, this policy also restricts the policies
to which a container may be spawned into. By default,
users will be able to transition to an unconfined
policy or any policy prefaced with 'docker-'.

Local operators may add new local policies prefaced
with 'docker-' without needing to modify this policy.
Operators choosing to disable privileged containers
will need to modify this policy to remove access
to change_policy to unconfined.

Signed-off-by: Eric Windisch <eric@windisch.us>

Commit:
daemon: container ArchivePath and ExtractToDir

The following methods will deprecate the Copy method and introduce
two new, well-behaved methods for creating a tar archive of a resource
in a container and for extracting a tar archive into a directory in a
container.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

Commit:
api/server: StatPath, ArchivePath, ExtractToDir

Adds http handlers for new API endpoints:

GET ContainersArchivePath
  Return a Tar Archive of the contents at the specified location in a
  container. Deprecates POST ContainersCopy. Use a HEAD request to stat
  the resource.

PUT ContainersExtractToDir
  Extract the Tar Archive from the request body to the directory at the
  specified location inside a container.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

Commit:
api/client: New and Improved `docker cp` behavior

Supports copying things INTO a container from a local file or from a tar
archive read from stdin.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

Commit:
integration-cli: New `docker cp` integration tests

Adds several integration tests for `docker cp` behavior with over a dozen
tests for each of:

  container -> local
  local -> container

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

Commit:
docs: Updated for docker cp and its API changes

Documented changes to API to enable new `docker cp` behavior.

Added documentation on `docker cp` usage and behavior.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

ewindisch committed:
Introduce a dedicated unconfined AA policy

By using the 'unconfined' policy for privileged
containers, we have inherited the host's apparmor
policies, which really make no sense in the
context of the container's filesystem.

For instance, policies written against
the paths of binaries such as '/usr/sbin/tcpdump'
can be easily circumvented by moving the binary
within the container filesystem.

Fixes GH#5490

Signed-off-by: Eric Windisch <eric@windisch.us>

Commit:
Windows: Statically linkable SQLite3

Signed-off-by: John Howard <jhoward@microsoft.com>

calavera committed:
Expose new mount points structs in inspect.

Keep old hashes around for old api version calls.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Define Docker's deprecation policy
Announce the cmd line options that are targetted for removal

Signed-off-by: Doug Davis <dug@us.ibm.com>

vbatts committed:
graph: preserve tar archive entries

Preserve the entries from the tar archive for layers added to the graph.

With these entries and relative filesystem path, the tar archives can be
reassembled later.

Signed-off-by: Vincent Batts <vbatts@redhat.com>

vbatts committed:
vendor: adding tar-split dependency for graph

tar-split is a facility to disassemble and reassemble tar archives

Signed-off-by: Vincent Batts <vbatts@redhat.com>

vbatts committed:
graph: variablize file names

and add a comment.. :-)

Signed-off-by: Vincent Batts <vbatts@redhat.com>

Conflicts:
	graph/graph.go

vbatts committed:
graph: use tar archive entries for TarLayer

if there is a tar-data.json.gz present for an image layer, then use it
to create the tar archive, instead of the traditional graphdriver Diff.

Signed-off-by: Vincent Batts <vbatts@redhat.com>

Conflicts:
	graph/graph.go

cpuguy83 committed:
Split reader interface from logger interface

Implement new reader interface on jsonfile.
Moves jsonlog decoding from daemon to jsonfile logger.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

cpuguy83 committed:
vendor fsnotify

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

NathanMcCauley committed:
Add cmdline docs for signing and verification flows

Signed-off-by: Nathan McCauley <nathan.mccauley@docker.com>

cpuguy83 committed:
remove dead code after decoupling from jsonlog

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

carlossg committed:
Clarify filters option in list containers and list images docs

Based on the list containers with filters options it would seem that filtering containers with label `test=docker-java` could be done with `{"test":["docker-java"]}` which doesn't work

The options that work are `{"label":["test"]}` and `{"label":["test=docker-java"]}`

As seen in https://github.com/docker-java/docker-java/pull/262

Signed-off-by: Carlos Sanchez <carlos@apache.org>

estesp committed:
Add links and correct locations in project/README.md

After removing the duplicate ROADMAP in a separate PR, a few other
issues were noted in README.md which are fixed here:
- the directory is project, not hack
- make.sh is no longer in the current dir since hack/ is not project/
  anymore
- MAINTAINERS is no longer here as a markdown file, but is a TOML file
  in the root of the project
- links were added to all files to make it easier to follow from here
  to the appropriate location

Docker-DCO-1.1-Signed-off-by: Phil Estes <estesp@linux.vnet.ibm.com> (github: estesp)

stefanberger committed:
Do not hide user provided network mounts [v2]

Prevent the docker daemon from mounting the created network files over
those provided by the user via -v command line option. This would otherwise
hide the one provide by the user.
The benefit of this is that a user can provide these network files using the
-v command line option and place them in a size-limited filesystem.

Signed-off-by: Stefan Berger <stefanb@us.ibm.com>

estesp committed:
Update docker-py to current master (~docker-py 1.3.1)

The docker-py commit used in the standard `Dockerfile` is from Feb. 2015
and is out of date with the current API level and has fixes for things
like the new docker cli config location and registry v2 changes/API
responses as well.

Also pass "NOT_ON_HOST=true" to docker-py test suite so that tests
relying on direct HOST interaction (versus running in a container) are
skipped.

Docker-DCO-1.1-Signed-off-by: Phil Estes <estesp@linux.vnet.ibm.com> (github: estesp)

calavera committed:
Promote volume drivers from experimental to master.

Remove volume stubs and use the experimental path as the only path.

Signed-off-by: David Calavera <david.calavera@gmail.com>

dmcgowan committed:
Use notary library for trusted image fetch and signing

Add a trusted flag to force the cli to resolve a tag into a digest via the notary trust library and pull by digest.
On push the flag the trust flag will indicate the digest and size of a manifest should be signed and push to a notary server.
If a tag is given, the cli will resolve the tag into a digest and pull by digest.
After pulling, if a tag is given the cli makes a request to tag the image.

Use certificate directory for notary requests

Read certificates using same logic used by daemon for registry requests.

Catch JSON syntax errors from Notary client

When an uncaught error occurs in Notary it may show up in Docker as a JSON syntax error, causing a confusing error message to the user.
Provide a generic error when a JSON syntax error occurs.

Catch expiration errors and wrap in additional context.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

jessfraz committed:
update docs for new install script

Signed-off-by: Jessica Frazelle <princess@docker.com>

estesp committed:
ps --format: Add config.js doc, fix gofmt, add integration tests

Re-add the docs from @calavera's PR to the moved cli cmd reference docs.
Fix gofmt and vet issues from carried commits
Add integration test for using format with --no-trunc and multi-names
Fix custom_test map order dependency on expected value check
Add docs to reference/commandline/ps.md
Remove "-F" flag option from original carried PR content

Docker-DCO-1.1-Signed-off-by: Phil Estes <estesp@linux.vnet.ibm.com> (github: estesp)

charleswhchan committed:
* Add benefit to using automated build.
* Wording based on suggestions by @thaJeztah, @moxiegirl in PR #14700.

Signed-off-by: Charles Chan <charleswhchan@users.noreply.github.com>

Commit:
Document branches and tags

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

vdemeester committed:
Update documentation according to opts updates

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

jessfraz committed:
update install script for new repos

Signed-off-by: Jessica Frazelle <princess@docker.com>

dmcgowan committed:
Add integration cli trust tests

Added notary server to docker base image.
Created trust suite which runs trust server for running trusted commands.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

LK4D4 committed:
Add docstring to reexec.Command

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

albers committed:
Second level completions for --log-opt

Advanced completion for some log driver options:
gelf-address, syslog-address, syslog-facility.

Signed-off-by: Harald Albers <github@albersweb.de>

Commit:
Refactor api/client/build.go

Separated preparation of context and Dockerfile for
the various different methods of specifying them.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

thaJeztah committed:
Merge pull request #14768 from fredlf/style-guide-edits

Small edits and clean up

Commit:
Merge pull request #14771 from Microsoft/10662-image2image

Fixing Image struct to no longer use Graph.

runcom committed:
move nat tests from container's unit test to nat's ones

Signed-off-by: Antonio Murdaca <runcom@linux.com>

MHBauer committed:
golint for cliconfig

 - fully capitalize HTTP in HTTPHeaders
 - comment for CONFIGFILE
 - camelcase and privatize oldConfigfile, defaultIndexserver
 - remove unused var errConfigFileMissing
 - comments for methods and functions throughout
 - external references to renamed variables changed

Signed-off-by: Morgan Bauer <mbauer@us.ibm.com>

runcom committed:
Merge pull request #14691 from Microsoft/10662-start

Windows: hostconfig on start

thaJeztah committed:
Remove reference to CENTOS6 from generate script

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Commit:
Add better output for TestLinksEtcHostsContentMatch to help debug #14774

Signed-off-by: Doug Davis <dug@us.ibm.com>

duglin committed:
Merge pull request #14775 from runcom/move-nat-tests

move nat tests from container's unit test to nat's ones

runcom committed:
Merge pull request #14776 from MHBauer/cliconfig-lint

golint fixes for cliconfig

duglin committed:
Merge pull request #14762 from LK4D4/lint_reexec

Add docstring to reexec.Command

tiborvass committed:
Merge pull request #14779 from duglin/BetterErrMsg14774

Add better output for TestLinksEtcHostsContentMatch to help debug #14774

vbatts committed:
graph: isolate the (dis)assembly logic

with the current duplication of code in the grap.go split-up, this puts
all assembly/disassembly logic into isolated functions

Signed-off-by: Vincent Batts <vbatts@redhat.com>

Commit:
Add golint to the development toolbox

Add golint to the Dockerfile, and a `validate-lint` task to the
Makefile. Currently, the linter will process a harcoded list of packages
that will expand as we fix more warnings. Eventually, the linter should
process all subpackages of the repo (excluding vendored code).

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

SvenDowideit committed:
Merge pull request #14757 from rhatdan/man

Fix man pages

thaJeztah committed:
Merge pull request #14744 from gesellix/add-gradle-docker-remote-api-plugin

Add gradle docker remote api plugin

thaJeztah committed:
Merge pull request #14766 from runcom/remove-ref-from-libcontainer

Remove reference to docker/libcontainer

WeiZhang555 committed:
golint on daemon/network package

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

thaJeztah committed:
Merge pull request #14727 from vdemeester/opts-docs-updates

Update documentation according to opts updates (#13694)

coolljt0725 committed:
Minor fix of TestLogsStderrInStdout

Signed-off-by: Lei Jitang <leijitang@huawei.com>

coolljt0725 committed:
Docs: update the devicemapper default basesize from 10G to 100G

Signed-off-by: Lei Jitang <leijitang@huawei.com>

rhatdan committed:
If a user hits this error it would be helpful to know tagstore name.

There are several bug reports on this error happening, and error is
not helpful unless you read the code.  Google brings up removing
the repositories.btrfs file.

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

cpuguy83 committed:
Merge pull request #14798 from coolljt0725/minor_fix_TestLogsStderrInStdout

Minor fix of TestLogsStderrInStdout

Commit:
golint: Fix issues in pkg/nat

Updates #14756

Signed-off-by: Dave Tucker <dt@docker.com>

thaJeztah committed:
Merge pull request #14731 from Evalle/14729-Fix-ubuntu-doc-issue

Fixing ubuntu doc issue

Commit:
golint: Lint pkg/namesgenerator

Also addded a couple more tests

Updates #14756

Signed-off-by: Dave Tucker <dt@docker.com>

Commit:
Merge pull request #14778 from thaJeztah/remove-rhel6-rpm-from-generate

Remove reference to CENTOS6 from generate script

cpuguy83 committed:
Bump go-md2man to 1.0.3

Fixes an issue with curly braces being escaped when they should not be.
This was particularly an issue in places where `--format '{{ ...  }}'`
is used in the man docs.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Merge pull request #14725 from charleswhchan/issue-14433

Issue #14433

runcom committed:
Merge pull request #14796 from WeiZhang555/golint-network

golint on daemon/network package

LK4D4 committed:
Merge pull request #14609 from ewindisch/apparmor-policy

Move AppArmor policy to contrib & deb packaging

cpuguy83 committed:
Merge pull request #14765 from runcom/fix-dockerCmd-refactor

Refactor missed dockerCmd changes

LK4D4 committed:
Fix cgroup parent tests

After merging mounting cgroups in container this tests doing wrong
checks. Cgroup paths could be prepended by other cgroups from host.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

runcom committed:
Merge pull request #14801 from HuKeping/golint

golint: use golint to check package util

LK4D4 committed:
Merge pull request #14780 from icecrime/14756_add_golint

Add golint to the development toolbox

LK4D4 committed:
Remove unused TtyTerminal interface

It was used only by integration tests, which now gone.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Merge pull request #13617 from Microsoft/10662-sql

Windows: Statically linkable SQLite3

runcom committed:
Merge pull request #14815 from LK4D4/remove_tty_terminal

Remove unused TtyTerminal interface

calavera committed:
Merge pull request #14659 from calavera/promote_volumes_experimental_to_master

Promote volume drivers from experimental to master.

runcom committed:
pkg: mount: golint

Fix the following warnings:

pkg/mount/mountinfo.go:5:6: type name will be used as mount.MountInfo by other packages, and that stutters; consider calling this Info
pkg/mount/mountinfo.go:7:2: struct field Id should be ID

Signed-off-by: Antonio Murdaca <runcom@linux.com>

HuKeping committed:
CI: use dockerCmd in integration-cli when possible

Signed-off-by: Hu Keping <hukeping@huawei.com>

Commit:
Remove dead code in registry package

The only uses of RequestAuthorization and its associated functions were
removed in 19515a7ad859b28c474d81e756ac245afcd968e3 ("Update graph to
use vendored distribution client for the v2 codepath")

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

stevvooe committed:
Adding Go coding style guidelines

Signed-off-by: Stephen J Day <stephen.day@docker.com>

Commit:
Remove unused types in registry package

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

runcom committed:
Allow starting a container with an existing hostConfig which contains links

Signed-off-by: Antonio Murdaca <runcom@linux.com>

Commit:
Improve documentation and golint compliance of registry package

* Add godoc documentation where it was missing

* Change identifier names that don't match Go style, such as INDEX_NAME

* Rename RegistryInfo to PingResult, which more accurately describes
  what this structure is for. It also has the benefit of making the name
  not stutter if used outside the package.

Updates #14756

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

Commit:
Merge pull request #14805 from dave-tucker/golint_namesgen

golint: Lint pkg/namesgenerator

Commit:
Vendor latest distributions so we can use the new errcode stuff

ping @stevvooe @RichardScothern

Signed-off-by: Doug Davis <dug@us.ibm.com>

rscothern committed:
Configure TLS for private registry mirrors.

If a registry mirror is using TLS, ensure that certs for it
are picked up from /etc/docker/certs.d

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

HuKeping committed:
golint: use golint to check package util

Signed-off-by: Hu Keping <hukeping@huawei.com>

Commit:
Windows: Remove erroneous comment

Signed-off-by: John Howard <jhoward@microsoft.com>

tiborvass committed:
Merge pull request #14820 from aaronlehmann/registry-dead-code

Remove dead code in registry package

Commit:
Add notary integration to `docker build`

The Dockerfile is rewritten with images references on FROM
instructions resolved to trusted digests. The rewritten Dockerfile
is swapped with the original one during context upload.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

Commit:
Update validate-lint script

Add a set of newly linted packages, and fix the script.

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

rscothern committed:
Remove v1 registry mirror configuration from LookupEndpoints.

V1 mirrors do not mirror the index and those endpoints should
only be indexes.

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

runcom committed:
Merge pull request #14830 from Microsoft/10662-removebadcomment

Windows: Remove erroneous comment only (no code change)

calavera committed:
Merge pull request #14682 from duglin/Issue14621

Remove panic in nat package on invalid hostport

Commit:
Do Docker edits so we can use the new distribution code

Signed-off-by: Doug Davis <dug@us.ibm.com>

tiborvass committed:
Merge pull request #14829 from RichardScothern/registry-tls

Configure TLS for private v2 registry mirrors.

tiborvass committed:
Merge pull request #14833 from icecrime/14756_update_lint_script

Update validate-lint script

Commit:
Merge pull request #13171 from jlhawn/archive_copy

docker cp to and from containers

jessfraz committed:
better i/o timeout error on pull

Signed-off-by: Jessica Frazelle <princess@docker.com>

clintkitson committed:
added REX-Ray as volume plugin to plugin docs

Signed-off-by: Clinton Kitson <clintonskitson@gmail.com>

hqhq committed:
Remove unused parameter in NewTtyConsole

It's introduced in
68ba5f0b69c9f38 (Execdriver implementation on new libcontainer API)

But I don't see reson why we need it.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

Commit:
Merge pull request #14808 from cpuguy83/bump_md2man

Bump go-md2man to 1.0.3

Commit:
Merge pull request #14812 from LK4D4/fix_cgroup_parent_tests

Fix cgroup parent tests

Commit:
Merge pull request #14688 from Microsoft/10662-hookupmac

Windows: Hook up user supplied MAC

Commit:
Merge pull request #14677 from vincentbernat/fix/zsh-completion-update-4

zsh: update zsh completion for docker command

Commit:
Merge pull request #14794 from HuKeping/ci

CI: use dockerCmd in integration-cli when possible

Commit:
Merge pull request #14770 from albers/completion-log-opt

Bash completion for log drivers and their options

Commit:
Merge pull request #14822 from runcom/host-config-links-on-start

Allow starting a container with an existing hostConfig which contains links

diogomonica committed:
Add more integration tests for trusted push and pull

Signed-off-by: Diogo Monica <diogo@docker.com>

Commit:
Merge pull request #14804 from dave-tucker/golint_nat

golint: Fix issues in pkg/nat

LK4D4 committed:
Merge pull request #14842 from hqhq/hq_remove_rootuid

Remove unused parameter in NewTtyConsole

hqhq committed:
Fix golint warnings for builder

Addresses: #14756

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

thaJeztah committed:
Merge pull request #13711 from calavera/version_volumes_inspect

Expose new mount points structs in inspect.

thaJeztah committed:
Merge pull request #14821 from stevvooe/coding-style-guidelines

Adding Go coding style guidelines

runcom committed:
Fix wrong Content-Type returned by /images/search API

/images/search was replying with Content-Type text/plain instead
of application/json.
Fix #14846

Signed-off-by: Antonio Murdaca <runcom@linux.com>

rhatdan committed:
Merge branch 'master' of github.com:docker/docker into error

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

bfirsh committed:
Fix golint warnings for integration-cli

Signed-off-by: Ben Firshman <ben@firshman.co.uk>

duglin committed:
Merge pull request #14850 from runcom/14846-fix-get-images-search-content-type

Fix wrong Content-Type returned by /images/search API

duglin committed:
Merge pull request #14802 from rhatdan/error

If a user hits this error it would be helpful to know tagstore name.

cpuguy83 committed:
Merge pull request #14817 from runcom/golint-pkg-mount

pkg: mount: golint

rhvgoyal committed:
docker: Unmount -init layer root before taking a snapshot

When we are creating a container, first we call into graph driver to take
snapshot of image and create root for container-init. Then we write some
files to it and call into graph driver again to create container root
from container-init as base.

Once we have written files to container-init root, we don't unmount it
before taking a snapshot of it. Looks like with XFS it leaves it in such
a state that when we mount the container root, it goes into log recovery
path.

Jul 22 10:24:54 vm2-f22 kernel: XFS (dm-6): Mounting V4 Filesystem
Jul 22 10:24:54 vm2-f22 kernel: XFS (dm-6): Starting recovery (logdev: internal)
Jul 22 10:24:54 vm2-f22 kernel: XFS (dm-6): Ending recovery (logdev: internal)

This should not be required. So let us unmount container-init before use
it  as a base for container root and then XFS does not go into this
internal recovery path.

Somebody had raised this issue for ext4 sometime back and proposed the same
change. I had shot it down at that point of time. I think now time has
come for this change.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

LK4D4 committed:
Merge pull request #14790 from hqhq/hq_golint_build

Fix golint warnings for builder

tiborvass committed:
Merge pull request #14851 from bfirsh/golint-integration-cli

Golint integration-cli

Commit:
Update linted package list

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Merge pull request #14217 from ankushagarwal/ubuntu-15.04

Add support for 15.04, add systemd note for 15.04

NathanMcCauley committed:
Add trust tests for Docker create, run, push, and pull

Created date util function

Signed-off-by: Nathan McCauley <nathan.mccauley@docker.com>

azurezk committed:
#14474 skip DockerSuite.TestRunCapAddCHOWN on lxc

Signed-off-by: Zhang Kun <zkazure@gmail.com>

vdemeester committed:
Update validate-lint to lint all go files

By default, using go with package will only validate the go file for the
current platform (or at last misses file_windows.go for example). This
tries to fix that.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

calavera committed:
Merge pull request #14858 from icecrime/14756_update_linted_packages

Update linted package list

calavera committed:
Merge pull request #14699 from estesp/docker-ps-format

Carry #10255: Docker ps format

calavera committed:
Update init scripts to use `docker daemon`.

Signed-off-by: David Calavera <david.calavera@gmail.com>

cpuguy83 committed:
Merge pull request #14753 from flavio/zfs-initialization-errors

ZFS driver: raise better errors during init

diogomonica committed:
Add test for incorrect nonroot passphrase

Fix failing tests for create, push, and pull

Signed-off-by: Diogo Monica <diogo@docker.com>

calavera committed:
Merge pull request #14442 from cpuguy83/refactor_logdrvier_reader

Refactor log driver reader

calavera committed:
Remove duplicated code parsing parameters for the archiving handlers.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Update docs to use the new `docker daemon` command

Signed-off-by: Tibor Vass <tibor@docker.com>

LK4D4 committed:
Merge pull request #14628 from estesp/update-docker-py

Update docker-py commit to current master (~1.3.1 release)

calavera committed:
Merge pull request #14067 from vbatts/vbatts-tar-split

graph: layer tar disassembly and reassembly

jessfraz committed:
move leeroy under docker

Signed-off-by: Jessica Frazelle <princess@docker.com>

calavera committed:
Merge pull request #14871 from jfrazelle/change-leeroy-uri

move leeroy under docker

jessfraz committed:
actually update deb dockerfiles

Signed-off-by: Jessica Frazelle <princess@docker.com>

thaJeztah committed:
Merge pull request #14799 from coolljt0725/docs_fix_devicemapper_default_basesize

Docs: update the devicemapper default basesize from 10G to 100G

thaJeztah committed:
Merge pull request #14769 from clintonskitson/add_plugindocs_rexray

updated plugin docs to include REX-Ray

cpuguy83 committed:
Merge pull request #14613 from stefanberger/nohidevols2

Do not hide user provided network mounts [v2]

LK4D4 committed:
Merge pull request #14869 from calavera/remove_duplicated_cp_handlers_init

Remove duplicated code parsing parameters for the archiving handlers.

jfrazelle committed:
Merge pull request #14872 from jfrazelle/fix-debs-apparmour

 update deb dockerfiles

jfrazelle committed:
Merge pull request #14864 from ewindisch/apparmor-engine-policy

Add AppArmor policy for the docker binary

thaJeztah committed:
Merge pull request #14528 from estesp/add-readme-links

Add links and correct locations in project/README.md

calavera committed:
Merge pull request #14839 from jfrazelle/better-io-error

better i/o timeout error on pull

thaJeztah committed:
Merge pull request #13900 from duglin/Deprecate

Define Docker's deprecation policy

calavera committed:
Add -d to the deprecated feature list for 1.8.

Signed-off-by: David Calavera <david.calavera@gmail.com>

LK4D4 committed:
Merge pull request #14862 from vdemeester/validate-lint-all-files

Update validate-lint to lint all go files

Commit:
Enable validate-lint as part of CI

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

jfrazelle committed:
Merge pull request #14638 from jfrazelle/new-repo-install-script

update install script/docs for new apt and yum repos

Commit:
Stop SIGSEGV on test dial not reachable

Signed-off-by: John Howard <jhoward@microsoft.com>

NathanMcCauley committed:
Add tests simulating an evil server

Signed-off-by: Nathan McCauley <nathan.mccauley@docker.com>

tiborvass committed:
Merge pull request #14580 from rohitkadam19/patch-1

Update docker_remote_api_v1.17.md

jfrazelle committed:
Merge pull request #14878 from icecrime/14756_golint_on_ci

Enable validate-lint as part of CI

Commit:
Avoid redundant HEAD requests for identical layers on push

pushV2Tag already deduplicates layers, but the scope of this
deduplication is only for a particular tag. If we are pushing all tags
in a repository, we may check layers several times. Fix this by moving
the layersSeen map from the pushV2Tag function to the v2Pusher struct.

In addition to avoiding some useless round-trips, this makes the "docker
push" output less confusing. It formerly could contain many repeated
lines like:

    124e2127157f: Image already exists
    124e2127157f: Image already exists
    ...

Add test coverage based on the "docker push" output: a hash should not
appear multiple times when pushing multiple tags.

Fixes #14873

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

stevvooe committed:
Merge pull request #14884 from aaronlehmann/redundant-image-checks

Avoid redundant HEAD requests for identical layers on push

Commit:
Fix typos in deprecation doc

thanks @tiborvass

Signed-off-by: Doug Davis <dug@us.ibm.com>

diogomonica committed:
Added tests for expired snapshots and timestamps

Signed-off-by: Diogo Monica <diogo@docker.com>

coolljt0725 committed:
Add ulimit to docker build.

Signed-off-by: Lei Jitang <leijitang@huawei.com>

Commit:
Windows: Stupid copy/paste error

Signed-off-by: John Howard <jhoward@microsoft.com>

estesp committed:
Merge pull request #12850 from coolljt0725/add_ulimi_to_build

Add ulimit to docker build

estesp committed:
Merge pull request #14886 from duglin/FixDepDoc

Fix typo in deprecation doc

estesp committed:
Merge pull request #14880 from Microsoft/10662-paniconremotetest

Stop SIGSEGV on test dial not reachable

jgeiger committed:
Fix typo in builder.mb .dockerignore example

Signed-off-by: jgeiger <jgeiger@gmail.com>

Evalle committed:
Fixing SLES12 issue

Signed-off-by: evalle <shmarnev@gmail.com>

dmcgowan committed:
Vendor notary

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

runcom committed:
Ensure body is closed after error is checked

Signed-off-by: Antonio Murdaca <runcom@linux.com>

albers committed:
Fix problem with = in completion of docker global options

Without this fix, `docker --log-driver fluentd --log-opt fluentd-tag=b`
would complete `b` to `build`.
Completion of the commands has to be nailed to __docker_pos_first_nonflag

Signed-off-by: Harald Albers <github@albersweb.de>

dmcgowan committed:
Add build integration cli tests

Clean up tests to remove duplicate code

Add tests which run pull and create in an isolated configuration directory.
Add build test for untrusted tag

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

albers committed:
Complete all known driver options if no log driver was specified

Without this fix, `docker --log-opt ` would not complete anything
because the completions were driver specific.

Signed-off-by: Harald Albers <github@albersweb.de>

Commit:
Add filter to /images/json docs

Closes #14894

Signed-off-by: Doug Davis <dug@us.ibm.com>

estesp committed:
Merge pull request #14898 from runcom/sockRequestRaw-close-body

Ensure body is closed after error is checked

jessfraz committed:
pass DOCKER_EXPERIMENTAL into build images

Signed-off-by: Jessica Frazelle <princess@docker.com>

jessfraz committed:
fix fedora release

Signed-off-by: Jessica Frazelle <princess@docker.com>

Commit:
Merge pull request #14905 from jfrazelle/fix-fedora-install

fix fedora release

thaJeztah committed:
Merge pull request #14893 from Evalle/14653-fix-sles-issue

Fixing SLES12 issue

thaJeztah committed:
Merge pull request #14891 from jgeiger/fix_typo_in_builder_docs_for_dockerignore

Fix typo in builder.mb .dockerignore example

calavera committed:
Merge pull request #14901 from albers/completion-log-opt-fixes

Fixes to bash completion for log driver options

jfrazelle committed:
Merge pull request #14903 from jfrazelle/fix-experimental-rpm-debs

pass DOCKER_EXPERIMENTAL into build images

calavera committed:
Merge pull request #14855 from ewindisch/apparmor-unconfined

Introduce a dedicated unconfined AA policy

tianon committed:
Merge pull request #13933 from ypid/mkimage-no-compression

Do not compress rootfs archive in mkimage.sh.

estesp committed:
Merge pull request #14887 from Microsoft/10662-stupidtypo

Windows: Stupid copy/paste error

Commit:
Merge pull request #14900 from duglin/AddFilterToDoc

Add filter to /images/json docs

vbatts committed:
Merge pull request #14854 from rhvgoyal/unmount-base

docker: Unmount -init layer root before taking a snapshot

LK4D4 committed:
Merge pull request #14034 from hqhq/hq_checkconfig_xattr

Some check_config enhancement

stevvooe committed:
Merge pull request #14831 from duglin/VendorDist

Vendor latest distributions so we can use the new errcode stuff

tianon committed:
Fix "docker ps" with no containers regression

The header row was not being printed when "docker ps" was invoked without containers thanks to the new format support, and we instead received a single blank line.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

LK4D4 committed:
Merge pull request #14545 from Djelibeybi/oraclelinux6-rpm

Updated docker-engine.spec to build on Oracle Linux 6.

jessfraz committed:
warn the script is depreciated

Signed-off-by: Jessica Frazelle <princess@docker.com>

Djelibeybi committed:
Switch to using only the RPM command to determine the distro version.

Signed-off-by: Avi Miller <avi.miller@oracle.com>
(cherry picked from commit a43199f14314b3d966e5723cd336f9747a0f1ca0)

calavera committed:
Remove reference to old Volumes field in inspect struct.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Djelibeybi committed:
Fix typo.

Signed-off-by: Avi Miller <avi.miller@oracle.com>
(cherry picked from commit 59e48b346848a131d85a808fa1e21aea8c4f9833)

dmcgowan committed:
Fix pruning on tar split

Currently the vendor script prunes files which are currently checked in.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

dmcgowan committed:
Keep license files

Currently the vendor script removes directories which do not have imported go packages, however this also ends up removing license files which may be other directories.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

Commit:
Merge pull request #14589 from paetling/ae-update_net_docs

add to docs that ports do not get exposed when using --net

Commit:
Merge pull request #14557 from maaquib/14418-dangling-link-in-runmd

Fixing dangling link in docs/reference/commandline/run.md

calavera committed:
Merge pull request #14919 from tianon/empty-ps-headers

Fix "docker ps" with no containers regression

djdefi committed:
Add index.md to examples docs

Signed-off-by: Ryan Trauntvein <rtrauntvein@novacoast.com>

jfrazelle committed:
Merge pull request #14926 from dmcgowan/prune-tar-split

Fix pruning in vendor script

Commit:
Merge pull request #14696 from rtrauntvein/patch-1

Fix link to examples documentation

stevvooe committed:
Impose moratorium on remote registry access

A moratorium has been added to the road map to add a moratorium on adding
remote registry access to commands that don't already do so.

Signed-off-by: Stephen J Day <stephen.day@docker.com>

jfrazelle committed:
Merge pull request #13771 from tiborvass/daemon-cli

New `docker daemon` command

djdefi committed:
Fix broken link in automated build doc

Signed-off-by: Ryan Trauntvein <rtrauntvein@novacoast.com>

thaJeztah committed:
Merge pull request #14700 from charleswhchan/patch-2

Add benefit to using automated build.

hqhq committed:
Remove unnecessary container cleanup in integration-cli

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

coolljt0725 committed:
Add VxFS magic number, fixes #14847

Signed-off-by: Lei Jitang <leijitang@huawei.com>

dmcgowan committed:
Rename to flags and environment variables to content trust

Update help line to allow 90 characters instead of 80

The trust flag pushes out the help description column wider, requiring more room to display help messages.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

tiborvass committed:
Merge pull request #14925 from calavera/fix_volume_symlink_test

Remove reference to old Volumes field in inspect struct.

thaJeztah committed:
Merge pull request #14877 from calavera/deprecate_docker_d

Add -d to the deprecated feature list for 1.8.

LK4D4 committed:
Fix option validation for log-drivers without it

There is no option validation for "journald" log-driver, so it makes no
sense to fail in that case.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

tianon committed:
Merge pull request #14865 from calavera/update_init_scripts

Update init scripts to use `docker daemon`.

calavera committed:
Merge pull request #14958 from LK4D4/fix_valid

Fix option validation for log-drivers without it

Commit:
reexec: Use in-memory binary on linux instead of os.Args[0]

This keeps reexec working properly even if the on-disk binary was replaced.

Signed-off-by: Tibor Vass <tibor@docker.com>

jfrazelle committed:
Merge pull request #14920 from jfrazelle/release-script-updates-for-new-repo

warn the script is depreciated

LK4D4 committed:
Merge pull request #14964 from tiborvass/fix-reexec

reexec: Use in-memory binary /proc/self/exe on linux instead of os.Args[0]

mrjana committed:
Vendoring libnetwork

Vendoring libnetwork commit f1c5671f1ee2133055144e566cd8b3a0ae4f0433

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

Commit:
Windows: Fixes panic on daemon binary

Signed-off-by: John Howard <jhoward@microsoft.com>

tianon committed:
Remove Ubuntu 14.10 (Utopic Unicorn) from build-deb targets

As of July 23, 2015, it is officially EOL (https://lists.ubuntu.com/archives/ubuntu-announce/2015-July/000198.html) and will receive no further updates.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

tiborvass committed:
Merge pull request #14835 from aaronlehmann/registry-lint-cleanup

Improve documentation and golint compliance of registry package

Commit:
Merge pull request #14939 from rtrauntvein/docs-autobuild

Fix broken link in automated build doc

duglin committed:
Merge pull request #14967 from Microsoft/10662-fixpaniconwindowsdaemon

Windows: Fixes panic on daemon binary

thaJeztah committed:
Merge pull request #14522 from carlossg/patch-1

Clarify filters option in list containers doc

thaJeztah committed:
Merge pull request #14637 from metalivedev/patch-1

Document that load supports compressed tarballs.

jessfraz committed:
fix memory swappiness lxc

Signed-off-by: Jessica Frazelle <princess@docker.com>

calavera committed:
Revert "Introduce a dedicated unconfined AA policy"

This reverts commit 87376c3add7dcd48830060652554e7ae43d11881.

Signed-off-by: David Calavera <david.calavera@gmail.com>

jfrazelle committed:
Merge pull request #14969 from tianon/utopic-eol

Remove Ubuntu 14.10 (Utopic Unicorn) from build-deb targets

jessfraz committed:
ignore certain tests on lxc driver

Signed-off-by: Jessica Frazelle <princess@docker.com>

jfrazelle committed:
Merge pull request #14977 from jfrazelle/fix-lxc

fix lxc

Commit:
Merge pull request #14935 from stevvooe/registry-moratorium

Impose moratorium on remote registry access

Commit:
Merge pull request #14976 from calavera/revert_unconfined_aa_policy

Revert "Introduce a dedicated unconfined AA policy"

Commit:
Merge pull request #14546 from dmcgowan/trusted-notary-integration

Notary integration

runcom committed:
Fix TestRunNonRootUserResolvName flackiness

Signed-off-by: Antonio Murdaca <runcom@linux.com>

Commit:
Merge pull request #14966 from mrjana/vendor18

Vendoring libnetwork

calavera committed:
Bump version to 1.8.0

Signed-off-by: David Calavera <david.calavera@gmail.com>
(cherry picked from commit d9f7fd9493e42d34a4be640a8c24781b8d7b4726)

tiborvass committed:
Merge pull request #14979 from runcom/fix-flacky-TestRunNonRootUserResolvName

Fix TestRunNonRootUserResolvName flackiness

LK4D4 committed:
Merge pull request #14941 from hqhq/hq_cli_remove_cleanup

Remove unnecessary container cleanup in integration-cli

thaJeztah committed:
Merge pull request #14720 from icecrime/14662_document_branches_tags

Document branches and tags

Commit:
Clarify error message when container name is already in use. Signed-off-by: Gustav Sinder <gustav.sinder@gmail.com>

Signed-off-by: gs11 <gustav.sinder@gmail.com>

icecrime commented:
The 1.8.0-rc1 **deb** and **rpm** packages are up for testing! See our [recent blog post](http://blog.docker.com/2015/07/new-apt-and-yum-repos/) for a description of the news apt & yum repositories.

### Install

`curl -sSL https://test.docker.com/ | sh`

### Version

```
$ docker version
Client:
 Version:      1.8.0-rc1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   4f1c66a
 Built:        Sat Jul 25 05:54:28 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.0-rc1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   4f1c66a
 Built:        Sat Jul 25 05:54:28 UTC 2015
 OS/Arch:      linux/amd64
```

One important change you may notice is that the distributed package is now a dynamically linked binary:

```
$ file $(which docker)
/usr/bin/docker: ELF 64-bit LSB  executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.24, BuildID[sha1]=df67b2dcebc1123765e0c55adb5a5438610f6894, not stripped
```

We will upload the usual static binaries on Monday. Happy testing!

thaJeztah commented:
@icecrime There was just an issue reported that the systemd units are missing; https://github.com/docker/docker/issues/14995.

thaJeztah committed:
Merge pull request #14245 from gs11/master

Clarifications in usage

chrisseto committed:
Fix go vet errors

Signed-off-by: Chris Seto <chriskseto@gmail.com>

jessfraz committed:
fix deb packaging systemd files

Signed-off-by: Jessica Frazelle <princess@docker.com>

thaJeztah committed:
Merge pull request #14940 from coolljt0725/add_vxfs_magic_number

Add VxFS magic number, fixes #14847

Commit:
Merge pull request #14996 from chrisseto/master

Fix go vet errors

runcom committed:
Format times in inspect command with a template as RFC3339Nano

In 1.6.2 we were decoding inspect API response into interface{}.
time.Time fields were JSON encoded as RFC3339Nano in the response
and when decoded into interface{} they were just strings so the inspect
template treated them as just strings.
From 1.7 we are decoding into types.ContainerJSON and when the template
gets executed it now gets a time.Time and it's formatted as
2015-07-22 05:02:38.091530369 +0000 UTC.
This patch brings back the old behavior by typing time.Time fields
as string so they gets formatted as they were encoded in JSON -- RCF3339Nano

Signed-off-by: Antonio Murdaca <runcom@linux.com>

calavera commented:
Closing this PR. We're going to use a more friendly release PR.
